modules/redis-lock/src/index.ts

```diff
@@ -0,0 +1,104 @@
+import { Node } from "@counterfactual/types";
+import Redis from "ioredis";
+import Redlock from "redlock";
+
+// async function delay(ms: number): Promise<void> {
+//   return new Promise(res => {
+//     setTimeout(() => res(), ms);
+//   });
+// }
+
+export class RedisLockService implements Node.ILockService {
+  private redlock: Redlock;
+
+  constructor(redisUrl: string) {
+    const redis = new Redis(redisUrl, {
+      retryStrategy: times => {
+        console.log("Lost connection to redis. Retrying to connect...");
+        const delay = Math.min(times * 50, 2000);
+        return delay;
+      },
+    });
+
+    this.redlock = new Redlock([redis], {
+      // the expected clock drift; for more details
+      // see http://redis.io/topics/distlock
+      driftFactor: 0.01, // time in ms
+
+      // the max number of times Redlock will attempt
+      // to lock a resource before erroring
+      retryCount: 100,
+
+      // the time in ms between attempts
+      retryDelay: 100, // time in ms
+
+      // the max time in ms randomly added to retries
+      // to improve performance under high contention
+      // see https://www.awsarchitectureblog.com/2015/03/backoff.html
+      retryJitter: 1000, // time in ms
+    });
+
+    this.redlock.on("clientError", err => {
+      console.error("A redis error has occurred:", err);
+    });
+  }
+
+  async acquireLock(
+    lockName: string,
+    callback: (...args: any[]) => any,
+    timeout: number,
+  ): Promise<any> {
+    const lockTTL = 10000;
+    timeout = lockTTL; // HACK-- switch bck to using given timeout
```

Comment:
Why not use given timeout here?

Reply:
Right now we don't know what CF node is passing in, let's confirm our params and then let them pass in their numbers.